### PR TITLE
Migrate mapper attachments plugin to main repository

### PR DESCRIPTION
**Pull request by rmuir**
We got this plugin synced up to master, improved tests, fixed securitymanager issues, added over 200 test documents from tika test suite, trimmed the parsers to a more contained list (e.g. word, excel, openoffice, powerpoint, pdf, html, rtf, txt, etc), and added special security restrictions to try to better contain the parsers.

I think its in pretty good shape and we should merge it into the main repository. The fact is, people use it, and even if we don't like the idea of parsing things server side, its currently something functional. Otherwise we have to maintain it in a separate repository and that has proved difficult. If we have a better solution in the future for dealing with these documents, we can still probably reuse the testing and tika handling and so on.

### CODE REVIEW

File: plugins/mapper-attachments/README.md [hunk 1]
```diff
@@ -0,0 +1,396 @@
+Mapper Attachments Type for Elasticsearch
```
Comment (dadoonet) [inline plugins/mapper-attachments/README.md]: I'll rewrite the doc in another commit later as a asciidoc in plugins doc.

**Comment by dadoonet** [comment]
Thank you Robert for doing this. It will help a lot for maintain this plugin in a good shape until we propose something else.
I'm sure that found team will appreciate it as well! :)

I'll take care of the documentation rewrite to asciidoc after this will be merged.

Do you think we could also merge it in 2.1?

**Comment by rmuir** [comment]
I set 2.2 because the issues i found were dealt with in ways that only 2.2+ can support.

**Comment by dadoonet** [comment]
Can we backport it to 2.1 without the fixes you wrote? I mean that the external repo won't have those fixes so we will have the same situation but at least the plugin will have the exact same lifecycle as elasticsearch itself?

**Comment by rmuir** [comment]
I'm not backporting bugs. This is a 3.0-only change now.

**Comment by rjernst** [comment]
LGTM

**Comment by rmuir** [comment]
Just to re-explain my reasoning for going to master only here. This PR is a three-pronged approach to fix security and insanity concerns:

1. whitelist parsers (requires master)
    This just simply cannot be done in maven, since it has no `transitive=false`. In gradle its easy:
https://github.com/rmuir/elasticsearch/blob/migrate-mapper-attachments/plugins/mapper-attachments/build.gradle#L28-L51. I know what the maven equivalent would be: unmaintainable.
2. sandbox parsers (requires 2.2+)
    We need security improvements in 2.2+ to further restrict the parsers so we have less concerns about bugs in them (especially stupid stuff like xml flaws). 
3. improve tests

So if gradle goes back to 2.2, then its safe to backport this there too. But without these 3 things its an incomplete solution.

**Pull request merged**